pkg/apiserver/middlewares/v1/api_key.go

```diff
@@ -46,22 +51,90 @@ func HashSHA512(str string) string {
 	return hashStr
 }
 
+func (a *APIKey) SetAllowedOUs(allowedOu []string) error {
+	for _, ou := range allowedOu {
+		//just drop empty strings, I guess ?
+		if ou == "" {
+			log.Warningf("Ignoring empty OU string in Bouncers Allowed OU (api-key)")
+			continue
+		}
+		a.AllowedOu = append(a.AllowedOu, ou)
+	}
+	log.Infof("%p Allowed Bouncers OU : %v", a, a.AllowedOu)
+	return nil
+}
+
 func (a *APIKey) MiddlewareFunc() gin.HandlerFunc {
```

Comment:
we need to honor the AuthType flag of the bouncer (do not allow a bouncer w/ AuthType set to "api key" to present CERT and vice-versa)